### PR TITLE
📝 Document how to subclass `Curator`

### PR DESCRIPTION
**Pull request by Zethson**
- Adds a guide on how to subclass the Curator base classes

Comes with:

- https://github.com/laminlabs/lamin-docs/pull/164

### CODE REVIEW

**Comment by falexwolf** [comment]
This looks already promising!

Often I wish that we'd delete one docs page for any new page added; but this here isn't such a case.

Small note: Can you please use `ln.track(uid)` instead of
image

**Comment by falexwolf** [comment]
Should one make it more explicit that this guide is about subclassing `Curator` to enable convenient curation of datasets of any format rather than "custom curation" of any dataset, the topic of https://docs.lamin.ai/curate-any?

To rephrase: The term "custom" may not be specific enough because there are two ways of custom curation. One via subclassing Curator (aka "high-level") and one via using the "low-level" lamindb API directly.

**Comment by Zethson** [comment]
Went more into the direction of `Subclassing Curator`. However, I don't know whether this header gives away how useful this feature is because I think that it's one of the best features of lamindb.

**Comment by falexwolf** [comment]
>  However, I don't know whether this header gives away how useful this feature is because I think that it's one of the best features of lamindb.

Then how about "Custom Curators" or similar and the "curate-any" guide is renamed to "Curate using the low-level registry API" or something?

**Comment by falexwolf** [comment]
Now starting a detailed review. Thanks for this, Lukas!

File: docs/curate.md [hunk 1]
```diff
@@ -5,4 +5,5 @@
 
 curate-df
 curate-any
+curate-subclass
```
Comment (falexwolf) [inline docs/curate.md:8]: If this is the recommended way to curate custom datasets we should put it first. Or we integrate it into curate-any and the curate-any guide starts with subclassing the Curator and only then walks through how to do it low-level.

3 documents on curation is potentially a bit intimidating in particular when curate-any will almost never be used and should be more like a footnote?

**Comment by falexwolf** [comment]
The previous commit refactored the changes made in this PR. Let me know if anything doesn't make sense, Lukas!

I'll now add another commit to simplify the docstring of `Curatator`:

image

We should then cross-link guides and find an answer to this question:

https://github.com/laminlabs/lamindb/pull/2022#discussion_r1802696408

**Comment by falexwolf** [comment]
Also added this:

- https://github.com/laminlabs/lamin-docs/pull/164

**Comment by falexwolf** [comment]
So, I made an attempt at the curator docstring, but this still throws me off: 

- https://github.com/laminlabs/lamindb/issues/1860

We can't have an easy explanation for what the Curator does if we don't have a simple definition of what validation is.

I think this PR here is a great opportunity to fix this once and for all.

File: lamindb/_curate.py [hunk 2]
```diff
@@ -950,29 +950,27 @@ def save_artifact(
...
-    - Values that can successfully validated and already exist in the registry.
-    - Values which are new and not yet validated or potentially problematic values.
-
-    3. Determine how to handle validated and non-validated values:
-
-    - Validated values not yet in the registry can be automatically registered using :meth:`~lamindb.core.DataFrameCurator.add_validated_from`.
-    - Valid and new values can be registered using :meth:`~lamindb.core.DataFrameCurator.add_new_from`.
-    - All unvalidated values can be accessed using :meth:`~lamindb.core.DataFrameCurator.non_validated` and subsequently removed from the object at hand.
+    - validated values not yet in the registry can be automatically registered using :meth:`~lamindb.core.DataFrameCurator.add_validated_from`
+    - new values found in the data can be registered using :meth:`~lamindb.core.DataFrameCurator.add_new_from`
```
Comment (falexwolf) [inline lamindb/_curate.py:972]: What is the difference between "new values" and "non-validated values"?

File: docs/curate.md (hunk 1 above)
Comment (Zethson) [reply docs/curate.md:8]: I think it is the recommended way if you want to enforce structure for a specific format and especially across people. However, if you're curating a dataset as a one off, subclassing `Curator` is too complex and annoying. Then the content in `curate-any` applies better I'd argue.

File: lamindb/_curate.py (hunk 2 above)
Comment (Zethson) [reply lamindb/_curate.py:972]: I think that the old text was not precise enough. The new one looks fine now.

**Comment by falexwolf** [comment]
@Zethson, I'm done here. I'd love for you to drive this home. 😇

**Comment by Zethson** [comment]
I decided to keep it as 3 documents for now instead of merging them and also showing this last. I'll keep a close eye on this and with a few days of reflection I expect myself to make a few small changes.

The `validated_from` issue we're aware of and it'll be fixed in subsequent PRs.

**Comment by falexwolf** [comment]
Sounds good! We'll polish more.

Re the very last commit: I don't like the `{doc}` syntax for the cellxgene ref because it shows the long title and hence makes the sentence & line very long. I know there is syntax to make the intersphinx work without showing the title, but I keep forgetting it. I'll fix this in another iteration.

**Pull request merged**